src/Scheduler.scala

```diff
@@ -0,0 +1,40 @@
+import java.util.concurrent.{Executors, PriorityBlockingQueue}
+
+import scala.concurrent.{ExecutionContext, Future}
+
+//type Task = () => Unit
+
+trait Scheduler {
+  def schedule(task: () => Unit, timestamp: Long): Unit
+  def start()(implicit ec: ExecutionContext): Unit
+}
+
+class SchedulerImp(
+    val minHeap: PriorityBlockingQueue[ScheduledTask],
+    // ExecutionContext for Future to execute a task in worker thread (not calling thread)
+    implicit val ec: ExecutionContext
```

Comment:
don't need `explicit` ExecutionContext here. being defined `implicit` ExecutionContext, the Future's ExecutionContext will use any ExecutionContext in the scope. In this case, because we instantiate an ExecutionContext in main method, this ec will be used.